### ISSUE TITLE
Auth Instance Type

### ISSUE THREAD

**Issue by jtlisi** [raintank/tsdb-gw#135]
Currently our authentication for cortex and graphite-shared will not ensure instance type the api key is authorized is for the correct instance type. This allows users to write to a cortex tenant with the same instance ID as a metrictank hosted metrics instance. The same would apply to graphite-shared.

Auth must be altered to verify the instance type.

**Comment by DanCech** [comment]
To be clear, the types are `graphite`, `graphite-shared` and `prometheus`.

**Comment by woodsaj** [comment]
Sounds like we just need a new arg in `auth/gcom/auth.go`
ie,
```
flag.StringVar(&validInstanceType, "auth-valid-instance-type", "", "if set, instance validation while fail if the type attribute of an instance does not match. (graphite|graphite-shared|prometheus)")
```

By default we can just "", which does no type validation.

**Comment by jtlisi** [comment]
Resolved with #136
